flink-connector/src/main/java/org/apache/flink/connector/delta/sink/SchemaConverter.java

```diff
@@ -0,0 +1,114 @@
+     * @return DeltaLake's specific type of stream's events
+     */
+    public StructType toDeltaFormat(RowType rowType) {
+        StructField[] fields = rowType.getFields()
+                .stream()
+                .map(rowField -> {
+                    DataType rowFieldType = toDeltaDataType(rowField.getType());
+                    return new StructField(
+                            rowField.getName(),
+                            rowFieldType,
+                            rowField.getType().isNullable());
+                })
+                .toArray(StructField[]::new);
+
+        return new StructType(fields);
+    }
+
+    /**
+     * Method containing the actual mapping between Flink's and DeltaLake's types.
+     *
+     * @param flinkType Flink's logical type
+     * @return DeltaLake's data type
+     */
+    public DataType toDeltaDataType(LogicalType flinkType) {
+        switch (flinkType.getTypeRoot()) {
+            case ARRAY:
+                org.apache.flink.table.types.logical.ArrayType arrayType =
+                        (org.apache.flink.table.types.logical.ArrayType) flinkType;
+                LogicalType flinkElementType = arrayType.getElementType();
+                DataType deltaElementType = toDeltaDataType(flinkElementType);
+                return new ArrayType(deltaElementType, flinkElementType.isNullable());
+            case BIGINT:
+                return new LongType();
+            case BINARY:
+            case VARBINARY:
+                return new BinaryType();
+            case BOOLEAN:
+                return new BooleanType();
+            case DATE:
+                return new DateType();
+            case DECIMAL:
+                org.apache.flink.table.types.logical.DecimalType decimalType =
+                        (org.apache.flink.table.types.logical.DecimalType) flinkType;
+                return new DecimalType(decimalType.getPrecision(), decimalType.getScale());
+            case DOUBLE:
+                return new DoubleType();
+            case FLOAT:
+                return new FloatType();
+            case INTEGER:
+                return new IntegerType();
+            case MAP:
+                org.apache.flink.table.types.logical.MapType mapType =
+                        (org.apache.flink.table.types.logical.MapType) flinkType;
+                DataType keyType = toDeltaDataType(mapType.getKeyType());
+                DataType valueType = toDeltaDataType(mapType.getValueType());
+                boolean valueCanContainNull = mapType.getValueType().isNullable();
+                return new MapType(keyType, valueType, valueCanContainNull);
+            case NULL:
+                return new NullType();
+            case SMALLINT:
+                return new ShortType();
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITH_TIME_ZONE:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return new TimestampType();
```

Comment:
similar question as above.

Reply:
same answer. I've put the same mapping for all the types because in `io.delta.standalone.types` there is only one `TimestampType` and its constructor does not accept any params.